mevolib/inference/_FastTree.py

```diff
@@ -1,142 +1,120 @@
+        infile_path: Input alignment file path.
+        bootstraps: Number of bootstraps to generate.
+        log_tmpfile: Path of the directory we want to save the FastTree output data into (Only required while testing for
+            reproducibility purposes or just if the user wants a specific folder to allocate the result of the execution
+            into. Otherwise, a random one will be provided).
 
     Returns :
-        list
-            List of arguments (excluding binary file) to call FastTree.
+        list: List of arguments (excluding binary file) to call FastTree.
     """
-    if ( args in KEYWORDS ) :
+    if args in KEYWORDS:
         argument_list = list(KEYWORDS[args])
-    else : # args not in KEYWORDS
-        argument_list = [arg  for arg in args.split(' ')]
+    else:  # args not in KEYWORDS
+        argument_list = [arg for arg in args.split(" ")]
     # Add the log file generation to get the log-likelihood score of the
     # resultant phylogeny
-    if ( '-log' not in argument_list ) :
-        log_tmpfile = tempfile.NamedTemporaryFile(delete=False)
-        argument_list += ['-log', log_tmpfile.name]
+    if "-log" not in argument_list:
+        if log_tmpfile is None:
+            log_tmpfile = tempfile.NamedTemporaryFile(delete=False).name
+        argument_list += ["-log", log_tmpfile]
     # Add the bootstrapping generation option if 'boostraps' is greater than 0
-    if ( bootstraps > 0 ) :
-        argument_list += ['-boot', str(bootstraps)]
+    if bootstraps > 0:
+        argument_list += ["-boot", str(bootstraps)]
     # Add the input file option
     argument_list += [infile_path]
-    return ( argument_list )
-
+    return argument_list
 
 
-def get_results ( command, output ) :
+def get_results(command: list, output: str) -> tuple[Bio.Phylo.BaseTree, float]:
     """
     Extract resultant phylogeny and its log-likelihood score from 'output' and
     files generated during the execution of 'command'.
 
     Arguments :
-        command  ( list )
-            FastTree's command line executed.
-        output  ( string )
-            Output from 'command' execution.
+        command: FastTree's command line executed.
+        output:  Output from 'command' execution.
 
     Returns :
-        Bio.Phylo.BaseTree
-            Resultant phylogenetic tree.
-        float
-            Log-likelihood score of the phylogeny.
+        Bio.Phylo.BaseTree: Resultant phylogenetic tree.
+        float: Log-likelihood score of the phylogeny.
     """
-    phylogeny = Phylo.read(StringIO(output), 'newick')
+
+    phylogeny = Phylo.read(StringIO(output), "newick")
     # Read the log file to get the log-likelihood score of the final phylogeny
-    index = command.index('-log') + 1
-    logfile_path = get_abspath(command[index])
-    with open(logfile_path, 'r') as logfile :
+    index = command.index("-log") + 1
+    logfile_path = Path(command[index]).absolute()
+    with open(logfile_path, "r") as logfile:
```

Comment:
```suggestion
    logfile_path = Path(command[index])
    with logfile_path.open("r") as logfile:
```
`Path()` allows you use `.open()`, which is a nice feature.